File: android/src/main/kotlin/com/shuzhi/direct_call/utils/CallRecord.kt
package com.shuzhi.direct_call.utils

data class CallRecord(
    val number: String,
    val type: Int,
    val date: Long,
    val duration: Long
) {
    // 将 CallRecord 转换为 Map
    fun toMap(): Map<String, Any> {
        return mapOf(
            "number" to number,
            "type" to type,
            "date" to date,
            "duration" to duration
        )
    }
}

File: lib/src/utils/call_record.dart

class CallRecord {
  String number;
  int type;
  int date;
  int duration;

  CallRecord({
    required this.number, 
    required this.type,
    required this.date,
    required this.duration
  });

  static fromMap(event) {
    return CallRecord(
      number: event['number'],
      type: event['type'],
      date: event['date'],
      duration: event['duration']
    );
  }
}